pulumi-deck/src/PulumiJobManifestPodLogs.tsx

```diff
@@ -0,0 +1,56 @@
+import { IInstanceMultiOutputLog, InstanceReader, JobManifestPodLogs } from '@spinnaker/core';
+
+import classNames from 'classnames';
+
+import React from 'react';
+
+import './pod-logs.less';
+
+export class PulumiJobManifestPodLogs extends JobManifestPodLogs {
+  private timeoutId: NodeJS.Timeout;
+  private consoleOutputFetchIntervalMs = 5000;
+
+  public componentWillUnmount() {
+    clearTimeout(this.timeoutId);
+  }
+
+  private getResourceRegion(): string {
+    return this.props.location;
+  }
+
+  private getPodName(): string {
+    const { podNameProvider } = this.props;
+    return `pod ${podNameProvider.getPodName()}`;
+  }
+
+  public async onClick() {
+    await this.fetchConsoleOutput();
+    this.open();
+  }
+
+  public close() {
+    clearTimeout(this.timeoutId);
+    super.close();
+  }
+
+  private async fetchConsoleOutput() {
+    if (this.state.errorMessage) {
+      return;
+    }
+
+    const { account } = this.props;
+    const region = this.getResourceRegion();
+
+    try {
+      const response = await InstanceReader.getConsoleOutput(account, region, this.getPodName(), 'kubernetes');
+      this.setState({
+        containerLogs: response.output as IInstanceMultiOutputLog[],
+        selectedContainerLog: response.output[0] as IInstanceMultiOutputLog,
+      });
+
+      this.timeoutId = setTimeout(async () => await this.fetchConsoleOutput(), this.consoleOutputFetchIntervalMs);
```

Comment:
Unfortunately, no. The response type contains a couple of other properties besides the console output string itself. Nothing in the response from what I saw would allow for a clean exit. However, I did think about passing in the execution status as a prop to this component all the way from its parent component that has access to the stage. But to do that, I'd have to turn _this_ component from a mere extension of the `JobManifestPodLogs` component to a full re-implementation of that component, just so that we can modify the props it accepts. If there are other ways to add a new prop I'd love to know! From my investigation there wasn't.

Also I did try to see if I could use some other service from `@spinnaker/core` that would allow me to query the pod execution status but I couldn't find one easily. I am hoping @danielpeach might know of a better way here.